angular/projects/spark-angular/schematics/ng-add/index.ts

```diff
@@ -1,36 +1,180 @@
+        `Could not set up "${browserAnimationsModuleName}" ` +
+        `because "${noopAnimationsModuleName}" is already imported.`,
+      );
+      context.logger.info(`Please manually set up browser animations.`);
+      return;
+    }
 
-    _context.addTask(new RunSchematicTask('ng-add-setup-project', _options), [installTaskId]);
+    addModuleImportToRootModule(tree, browserAnimationsModuleName, browserAnimationsModuleImport, project);
 
     return tree;
   };
 }
 
-function addDependencies(tree: Tree): Tree {
-  const dependencies: NodeDependency[] = [
-    { name: '@sparkdesignsystem/spark', overwrite: true, type: NodeDependencyType.Dev, version: '^13.0.0' },
-    { name: '@sparkdesignsystem/spark-angular', overwrite: true, type: NodeDependencyType.Dev, version: '^10.0.0' },
-  ];
+/**
+ * Adds the Spark stylesheets to the project style file
+ */
+function addSparkStyles(options: Schema): Rule {
+  return (tree: Tree, context: SchematicContext) => {
+    const workspace = getWorkspace(tree);
+    const project = getProjectFromWorkspace(workspace, options.project);
+    const styleFilePath = getProjectStyleFile(project);
+    const logger = context.logger;
+
+    if (!styleFilePath) {
+      logger.error(`Could not find the default style file for this project.`);
+      logger.info(`Please manually import the Spark stylesheet in your CSS.`);
+      return;
+    }
+
+    const buffer = tree.read(styleFilePath);
+
+    if (!buffer) {
+      logger.error(`Could not read the default style file within the project (${styleFilePath})`);
+      logger.info(`Please manually import the Spark stylesheet in your CSS.`);
+      return;
+    }
+
+    const htmlContent = buffer.toString();
+
+    const insertion = `
+@import "node_modules/@sparkdesignsystem/spark/spark.scss"
+`;
+
+    if (htmlContent.includes(insertion)) {
+      return;
+    }
+
+    const recorder = tree.beginUpdate(styleFilePath);
 
-  dependencies.forEach((dependency) => addPackageJsonDependency(tree, dependency));
+    recorder.insertLeft(htmlContent.length, insertion);
+    tree.commitUpdate(recorder);
 
-  return tree;
+    return tree;
+  };
+}
+
+/**
+ * Adds the Spark class to the project body tag
+ */
+function addSparkClass(options: Schema): Rule {
+  return (tree: Tree, context: SchematicContext) => {
+    const workspace = getWorkspace(tree);
+    const project = getProjectFromWorkspace(workspace, options.project);
+    const projectIndexFiles = getProjectIndexFiles(project);
+    const logger = context.logger;
+
+    if (!projectIndexFiles.length) {
+      logger.error(`Could not find the index file for this project.`);
+      logger.info(`Please manually add the "${sparkAngularJavaScriptClass}" class to your HTML tag.`);
+      return;
+    }
+
+    projectIndexFiles.forEach((indexFilePath) => {
+      // This should go on <html> but it is easier for now to just use body
+      addBodyClass(tree, indexFilePath, sparkAngularJavaScriptClass);
```

Comment:
Hey @jtneal , our expectation is that this class is added to HTML element.

Reply:
I know, as the comment states here it was much easier to place on body because @angular/cdk supports it. I will need to submit a PR to @angular/cdk to add support for updating the HTML tag, or else we'd have to copy a bunch of @angular/cdk code into this project to accomplish it. I am curious what use-case requires the class on the html tag where it wouldn't work on the body tag? That doesn't seem to make sense.

https://sparkdesignsystem.com/installing-spark/angular doesn't mention this as an installation step any more - so we should probably add that back to the installation documentation as well. In addition, I would expect we'd want it added the starter example: https://github.com/sparkdesignsystem/spark-starter-angular/blob/master/src/index.html

Reply:
@jtneal I'm having trouble understanding if CDK is part of Material or if it's an agnostic library for custom components that Material just uses. We haven't really focused on Spark being compatible with other UI systems like Material. Either way, changing where that class is put causes API differences between our implementations and has the possibility of breaking hooks that teams have already created with that class on HTML. There's too much risk of a breaking change to our consumers if we just change it without first deprecating the old way and phasing it out.

I think we need to put this on hold until we can gauge the impact that changing the class from html to body would have downstream. I think this is addition is a win for us in the long run, but not something we can just change quickly. Obviously there's also the option of opening a PR against CDK to make it possible to add classes to the html instead, if that is something you have the time or desire to do.

Reply:
@afebbraro Is that an option for now? I think we identified that our docs are missing the instruction to place the class on HTML and that we need to add that back in. For this, could we remove the code for adding the class to the body and add a note in the instructions that says you need to add the class to the HTML element if you are using vanilla js code (I think that's the usecase?)

Reply:
Hey all! We could do that but you also need to add it for Angular. So, this wouldnt be a full setup for our users and would require them to do additional work. Which could be awesome that we are doing some of the work for people but we could end up confusing them. What do you think about that @yodasw16 ?

Reply:
@yodasw16 What do you think about what I said up above?

Reply:
@afebbraro I think that since none of our implementations automatically add the JS utility class (right?), leaving it out of this with explicit instructions in the README that say you still need to add it is just as user-friendly as everything else. I would propose that we drop the class from the Body, add instructions to the README, and open an issue on the Material repo asking them to add the ability to put a class on the HTML element.

Reply:
@yodasw16 our Vanilla implementation adds it automatically. Your proposal sounds good to me though!